clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -4256,6 +4256,7 @@ bool CompilerInvocation::ParseLangArgs(LangOptions &Opts, ArgList &Args,
 
       if (TT.getArch() == llvm::Triple::UnknownArch ||
           !(TT.getArch() == llvm::Triple::aarch64 || TT.isPPC() ||
+            TT.getArch() == llvm::Triple::spirv64 ||
```

Comment:
Honestly the more I see added to this big if statement the more I'm convinced it doesn't need to exist.

Reply:
Right now it's at least throwing an error if the triple doesn't match the list, so there might be some case where removing the error will break something, so I'd prefer to not introduce a bug in a possibly untested area.